solr/solrj/src/test/org/apache/solr/client/solrj/impl/Http2SolrClientTest.java

```diff
@@ -36,34 +36,32 @@
 import org.eclipse.jetty.util.ssl.SslContextFactory;
 import org.junit.Test;
 
-public class Http2SolrClientTest extends Http2SolrClientTestBase<Http2SolrClient.Builder> {
+public class Http2SolrClientTest extends Http2SolrClientTestBase {
 
   @Override
   protected String expectedUserAgent() {
     return "Solr[" + Http2SolrClient.class.getName() + "] 2.0";
   }
 
   @Override
-  protected <B extends HttpSolrClientBuilderBase> B builder(
-      String url, int connectionTimeout, int socketTimeout, Class<B> type) {
```

Comment:
I'm on the fence of you removing `Class<B> type` for the builder.  I *think* this would enable you to remove all those SuppressWarning lines.

Reply:
I will say that writing a generic builder in Java is no easy task!  Maybe I'm wrong on this but I do not think we can avoid unchecked casts here unless we require the user to pass in the target class as an additional parameter on each method.